core/src/main/scala/io/qbeast/core/transform/Transformer.scala

```diff
@@ -127,40 +118,19 @@ trait Transformer extends Serializable {
 /**
  * Empty ColumnStats
  */
-object NoColumnStats extends ColumnStats(Nil, Nil, 0, 0.0, Nil)
+object NoColumnStats extends ColumnStats(Nil, Nil)
 
 /**
  * Stores the stats of the column
- * @param min    min value of the column
- * @param max    max value of the column
- * @param count  number of values in the column
- * @param stddev standard deviation of the column
- * @param mean   mean of the column
+ * @param names the stats names
+ * @param columns the stats column operation
  */
-case class ColumnStats(min: Any, max: Any, count: Long, stddev: Any, mean: Any)
-    extends Serializable {}
-
-object ColumnStats {
-
-  def apply(stats: Map[String, String], dataType: QDataType): ColumnStats = {
-
-    val min = stats("min")
-    val max = stats("max")
-    val count = stats("count").toLong
-    val stddev = stats("stddev")
-    val mean = stats("mean")
-
-    val (minVal, maxVal) = dataType match {
-      case DoubleDataType => (min.toDouble, max.toDouble)
-      case IntegerDataType => (min.toInt, max.toInt)
-      case LongDataType => (min.toLong, max.toLong)
-      case FloatDataType => (min.toFloat, max.toFloat)
-      case DecimalDataType => (min.toDouble, max.toDouble)
-      case StringDataType => (Nil, Nil)
-      case _ => throw new RuntimeException(s"Data type $dataType not supported")
-    }
-
-    ColumnStats(minVal, maxVal, count, stddev, mean)
-  }
+case class ColumnStats(names: Seq[String], columns: Seq[String]) extends Serializable {
```

Comment:
that is a column? I guess is the type of statistic that we need? Like if a column is named age, we will get max_age, min_age, no? I think it would be best to call names columns, and columns stats ( or aggregation, SQL, metrics, whatever you like).